include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -297,20 +244,37 @@ template <typename _T, typename... _Events>
 using __enable_if_convertible_to_events =
     typename ::std::enable_if<oneapi::dpl::__internal::__is_convertible_to_event<_Events...>::value, _T>::type;
 
-// Extension: execution policies type traits
-template <typename _ExecPolicy, typename _T, typename... _Events>
-using __enable_if_device_execution_policy = typename ::std::enable_if<
-    oneapi::dpl::__internal::__is_device_execution_policy<typename ::std::decay<_ExecPolicy>::type>::value &&
-        oneapi::dpl::__internal::__is_convertible_to_event<_Events...>::value,
-    _T>::type;
 
-template <typename _ExecPolicy, typename _T>
-using __enable_if_hetero_execution_policy = typename ::std::enable_if<
-    oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<_ExecPolicy>::type>::value, _T>::type;
+template <typename _BackendTag>
+struct __offload_tag
+{
+    // __backend_tag to do not duplicate the algorithm_impl level
+    using __backend_tag = _BackendTag;
+};
 
-template <typename _ExecPolicy, typename _T>
-using __enable_if_fpga_execution_policy = typename ::std::enable_if<
-    oneapi::dpl::__internal::__is_fpga_execution_policy<typename ::std::decay<_ExecPolicy>::type>::value, _T>::type;
+struct __device_backend
+{
+};
+
+template <class... _IteratorTypes, typename _KernelName>
+typename ::std::enable_if<__is_random_access_iterator<_IteratorTypes...>::value, __offload_tag<__device_backend>>::type
```

Comment:
Do we want to rename `__device_backend` to `dpcpp_backend` and `__offload_tag` to `hetero_tag` ?

Reply:
Probably, I don't think this is the showstopper for the current review. The names could be considered later.

Reply:
I think it is important for the upcoming public discussion - to have consistent names with current name and terms. It will help better understand what we propose.

Reply:
I would prefer to keep the names of backend tags consistent with policies but I agree with changing `offload_tag` to `hetero_tag.